rustls/tests/api.rs

```diff
@@ -2029,26 +2029,31 @@ fn do_exporter_test(client_config: ClientConfig, server_config: ServerConfig) {
     );
     do_handshake(&mut client, &mut server);
 
-    assert_debug_eq(
-        client.export_keying_material(&mut client_secret, b"label", Some(b"context")),
-        Ok(()),
-    );
-    assert_debug_eq(
-        server.export_keying_material(&mut server_secret, b"label", Some(b"context")),
-        Ok(()),
+    let exported_client_secret_with_ctx = client
```

Comment:
It doesn't look like these changes are actually necessary? Without them, do you get an error? What does that look like?

Reply:
If I back out all the changes to `rustls/test/api.rs` then the assertions fail to compile w/ a type mismatch because the return type isn't `Ok(())` in the success case any longer, e.g.:
```
error[E0308]: mismatched types
    --> rustls/tests/api.rs:2034:12
     |
2034 |         Ok(()),
     |         -- ^^ expected `&mut [u8; 64]`, found `()`
     |         |
     |         arguments to this enum variant are incorrect
     |
note: tuple variant defined here`
```

I think the test asserts will have to change somehow, but my first pass might be too invasive.

<details>
<summary>Here's one potential alternative. WDYT of this diff? :</summary>
```diff
diff --git a/rustls/tests/api.rs b/rustls/tests/api.rs
index 341fe068..6d6edf85 100644
--- a/rustls/tests/api.rs
+++ b/rustls/tests/api.rs
@@ -2029,25 +2029,21 @@ fn do_exporter_test(client_config: ClientConfig, server_config: ServerConfig) {
     );
     do_handshake(&mut client, &mut server);

-    assert_debug_eq(
-        client.export_keying_material(&mut client_secret, b"label", Some(b"context")),
-        Ok(()),
-    );
-    assert_debug_eq(
-        server.export_keying_material(&mut server_secret, b"label", Some(b"context")),
-        Ok(()),
-    );
+    assert!(client
+        .export_keying_material(&mut client_secret, b"label", Some(b"context"))
+        .is_ok());
+    assert!(server
+        .export_keying_material(&mut server_secret, b"label", Some(b"context"))
+        .is_ok());
     assert_eq!(client_secret.to_vec(), server_secret.to_vec());

-    assert_debug_eq(
-        client.export_keying_material(&mut client_secret, b"label", None),
-        Ok(()),
-    );
+    assert!(client
+        .export_keying_material(&mut client_secret, b"label", None)
+        .is_ok());
     assert_ne!(client_secret.to_vec(), server_secret.to_vec());
-    assert_debug_eq(
-        server.export_keying_material(&mut server_secret, b"label", None),
-        Ok(()),
-    );
+    assert!(server
+        .export_keying_material(&mut server_secret, b"label", None)
+        .is_ok(),);
     assert_eq!(client_secret.to_vec(), server_secret.to_vec());
 }
```
</details>

Reply:
Yup, that looks good to me.

Reply:
Great :-) Applied in my force push of dc6bef766a0f017770f49aa4f37a9d2aebc14ebf.